### PR TITLE
Settings crd

### PR DESCRIPTION
**Pull request by harveyxia**

### CODE REVIEW

File: api/settings/v1alpha2/settings.proto [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+message SettingsSpec {
+
+    // Configure mTLS settings for TrafficTargets.
+    MTLS mtls = 1;
+
+    // mTLS settings.
+    message MTLS {
+
+        // If true, by default enable mTLS for all TrafficTargets unless explicitly disabled by TrafficPolicies.
+        bool default_mtls = 1;
```
Comment (joekelley) [inline api/settings/v1alpha2/settings.proto:18]: should this be as simple as on/off, or should we also allow for simple TLS? https://istio.io/latest/docs/reference/config/networking/destination-rule/#ClientTLSSettings-TLSmode

Comment (harveyxia) [reply api/settings/v1alpha2/settings.proto:18]: Good question—we currently only use `ISTIO_MUTUAL` for mTLS. @ilackarms any thoughts?

**Comment by harveyxia** [comment]
Closed in favor of https://github.com/solo-io/service-mesh-hub/pull/1004

File: api/settings/v1alpha2/settings.proto (hunk 1 above)
Comment (harveyxia) [reply api/settings/v1alpha2/settings.proto:18]: Discussed offline, we'll import all of the Istio TLS modes, [reference](https://istio.io/latest/docs/reference/config/networking/destination-rule/#ClientTLSSettings-TLSmode)

**Pull request closed without merging**